app/controllers/clearance/passwords_controller.rb

```diff
@@ -38,45 +37,50 @@ def update
   private
 
   def deliver_email(user)
-    mail = ::ClearanceMailer.change_password(user)
-    mail.deliver_later
+    ::ClearanceMailer.change_password(user).deliver_later
   end
 
   def password_reset_params
     params[:password_reset][:password]
   end
 
-  def find_user_by_id_and_confirmation_token
-    user_param = Clearance.configuration.user_id_parameter
-
-    Clearance.configuration.user_model.
-      find_by_id_and_confirmation_token params[user_param], params[:token].to_s
-  end
-
   def find_user_for_create
     Clearance.configuration.user_model.
       find_by_normalized_email params[:password][:email]
   end
 
+  def find_user_by_password_reset_token(token)
+    verifier = Clearance.configuration.message_verifier
+    user_model = Clearance.configuration.user_model
+
+    begin
+      user_id, encrypted_password, expiration = verifier.verify(token)
```

Comment:
Maybe `encrypted_password` shouldn't be part of what we sign here? It allows us to invalidate any tokens as soon as the password is changed but it also means the encrypted password is part of the payload (signed). I dont know that the tradeoff is worth it. Curious for additional thoughts here...